### PR TITLE
feature/SMHE-1962-postprocess-response-of-set-and-get-configuration-object-requests

### PR DESCRIPTION
**Pull request by harrymiddelburgetall**
After Getting and Setting Configuration Object hls3active, hls4active and hls5active attributes of the dlms device should be updated in the dlms adapter database.

### CODE REVIEW

**Comment by harrymiddelburgetall** [comment]
✔ https://jenkins.fdp.osgp.cloud/view/SMHE-cucumber/job/gxf-smhe-cucumber-build/198/

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/configuration/service/AbstractConfigurationObjectService.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.repositories.DlmsDeviceRepository;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ConfigurationFlagDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ConfigurationFlagTypeDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ConfigurationObjectDto;
+
+public abstract class AbstractConfigurationObjectService implements ConfigurationObjectService {
+
+  private final DlmsDeviceRepository dlmsDeviceRepository;
+
+  public AbstractConfigurationObjectService(final DlmsDeviceRepository dlmsDeviceRepository) {
```
Comment (natasja-n) [inline osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/configuration/service/AbstractConfigurationObjectService.java]: Abstract classes should not have public constructors. Constructors of abstract classes can only be called in constructors of their subclasses. So there is no point in making them public. The protected modifier should be enough.

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/configuration/service/GetConfigurationObjectService.java [hunk 2]
```diff
@@ -22,12 +24,19 @@
...
 import org.slf4j.LoggerFactory;
 
-public abstract class GetConfigurationObjectService implements ProtocolService {
+public abstract class GetConfigurationObjectService extends AbstractConfigurationObjectService {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(GetConfigurationObjectService.class);
 
+  protected DlmsDeviceRepository dlmsDeviceRepository;
+
+  public GetConfigurationObjectService(final DlmsDeviceRepository dlmsDeviceRepository) {
```
Comment (natasja-n) [inline osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/configuration/service/GetConfigurationObjectService.java]: Abstract classes should not have public constructors. Constructors of abstract classes can only be called in constructors of their subclasses. So there is no point in making them public. The protected modifier should be enough.

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/configuration/service/GetConfigurationObjectServiceDsmr43.java [hunk 3]
```diff
@@ -28,4 +35,18 @@ public boolean handles(final Protocol protocol) {
   Optional<ConfigurationFlagTypeDto> getFlagType(final int bitPosition) {
     return ConfigurationFlagTypeDto.getDsmr43FlagType(bitPosition);
   }
+
+  @Override
+  protected void addLowFlags(final List<ConfigurationFlagDto> configurationFlags) {
```
Comment (natasja-n) [inline osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/configuration/service/GetConfigurationObjectServiceDsmr43.java:40]: Ik zie niet waar addLowFlags geraakt wordt in de unit tests.

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/configuration/service/AbstractConfigurationObjectService.java (hunk 1 above)
Comment (harrymiddelburgetall) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/configuration/service/AbstractConfigurationObjectService.java]: constructor is now protected

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/configuration/service/GetConfigurationObjectService.java (hunk 2 above)
Comment (harrymiddelburgetall) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/configuration/service/GetConfigurationObjectService.java]: constructor is now protected

File: osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/configuration/service/GetConfigurationObjectServiceDsmr43.java (hunk 3 above)
Comment (harrymiddelburgetall) [reply osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/configuration/service/GetConfigurationObjectServiceDsmr43.java:40]: Added tests for three protocol versions. Adding the low flags is deep down so I had to do a lot of mocking.

**Review by natasja-n — APPROVED** [review]

**Pull request merged**